<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainDialog</class>
 <widget class="QDialog" name="MainDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>877</width>
    <height>789</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainDialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_4">
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QFrame" name="frame_4">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Pins Info</string>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
         <property name="checkable">
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="8" column="2">
           <widget class="QPushButton" name="pushButton_4">
            <property name="text">
             <string>Apply Changes</string>
            </property>
           </widget>
          </item>
          <item row="3" column="2">
           <widget class="QCheckBox" name="chkInput">
            <property name="text">
             <string>Input Pin</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0" rowspan="5">
           <widget class="QListWidget" name="listPins">
            <property name="selectionMode">
             <enum>QAbstractItemView::ExtendedSelection</enum>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QLineEdit" name="txtPinName"/>
          </item>
          <item row="4" column="2">
           <widget class="QPlainTextEdit" name="txtDesc"/>
          </item>
          <item row="2" column="2">
           <widget class="QComboBox" name="cmbDataType">
            <item>
             <property name="text">
              <string>BitChannel</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>BitAChannel</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Complex128Channel</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Complex128AChannel</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>FloatAChannel</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>FloatChannel</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QPushButton" name="pushButton">
            <property name="text">
             <string>Add Pin</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Module Info</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QListWidget" name="listModules"/>
          </item>
          <item>
           <widget class="QFrame" name="frame_5">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <item>
              <widget class="QPushButton" name="btnApplyModule">
               <property name="text">
                <string>Apply</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="txtModuleName">
               <property name="placeholderText">
                <string>Module Name</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Function Name</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="txtFunctionName"/>
             </item>
             <item>
              <widget class="QPushButton" name="pushButton_2">
               <property name="text">
                <string>Fill Input Pins</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="txtInPins"/>
             </item>
             <item>
              <widget class="QPushButton" name="pushButton_5">
               <property name="text">
                <string>Fill Output Pins</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="txtOutPins"/>
             </item>
             <item>
              <widget class="QLabel" name="label_8">
               <property name="text">
                <string>Code</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QTextEdit" name="textEdit">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>200</height>
                </size>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="verticalSpacer">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
            <zorder>pushButton_5</zorder>
            <zorder>txtModuleName</zorder>
            <zorder>label_2</zorder>
            <zorder>txtFunctionName</zorder>
            <zorder>txtInPins</zorder>
            <zorder>txtOutPins</zorder>
            <zorder>btnApplyModule</zorder>
            <zorder>textEdit</zorder>
            <zorder>label_8</zorder>
            <zorder>pushButton_2</zorder>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QFrame" name="frame_3">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="3" column="0">
        <widget class="QSpinBox" name="inPins">
         <property name="value">
          <number>1</number>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QPushButton" name="pushButton_3">
         <property name="text">
          <string>Fill Settings</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0" colspan="3">
        <widget class="QTextEdit" name="txtFullText"/>
       </item>
       <item row="6" column="1">
        <widget class="QToolButton" name="toolButton">
         <property name="text">
          <string>Generate JSON</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QSpinBox" name="outPins">
         <property name="minimum">
          <number>1</number>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="lineChipName">
         <property name="text">
          <string>SampleChip</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>ChipName</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QSpinBox" name="nModules">
         <property name="minimum">
          <number>1</number>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Modules </string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Input Pins</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Output Pins</string>
         </property>
        </widget>
       </item>
       <item row="8" column="0">
        <widget class="QPushButton" name="pushButton_6">
         <property name="text">
          <string>Save As</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
